### DISCUSSION TITLE
How do i install commit-message-linter

### DISCUSSION THREAD

**Discussion by rafaelxvr** [alirezanet/Husky.Net#77] [General]
The is a hook on the documentation that uses commit-message-linter, but it shows off an error

hook:
`    {
      "name": "commit-message-linter",
      "group": "pre-commit",
      "command": "dotnet",
      "args": [
        "husky",
        "exec",
        ".husky/csx/commit-lint.csx",
        "--args",
        "${args}"
      ]
    },`

error: `no arguments passed to run the command`

Do i need to install something?

**Comment by alirezanet** [comment]
you need to copy[ `commit-message-linter.csx` ](https://github.com/alirezanet/Husky.Net/blob/master/.husky/csx/commit-lint.csx)from this repo or create your own... (you can find it in the .husky/csx folder)
csx files are just scripts so it wouldn't install automatically.

**Reply by alirezanet** [reply]
and also make sure your `commit-msg` hook contains the arg:

e.g
```
#!/bin/sh
. "$(dirname "$0")/_/husky.sh"

dotnet husky run --name "commit-message-linter" --args "$1"
echo
echo Great work! 🥂
echo

```

**Reply by rafaelxvr** [reply]
@alirezanet can you give me a example how should my hook on task-runner.json should look like?

Currently is:

`{
      "name": "commit-message-linter",
      "group": "pre-commit",
      "command": "dotnet",
      "args": [
        "husky",
        "exec",
        ".husky/csx/commit-lint.csx",
        "--args",
        "${args}"
      ]
    }`

**Reply by alirezanet** [reply]
This repo is using the same configuration and commit message linter. so the best example is to check the files in `.husky` folder

**Comment by rafaelxvr** [comment]
@alirezanet i'm having an issue where the husky not runs before i do a commit on my project, do i need to add any configuration besides the hooks?

**Reply by rafaelxvr** [reply]
commit-msg.txt: 
`#!/bin/sh
. "$(dirname "$0")/_/husky.sh"

dotnet husky run --name "commit-message-linter" --args "$1"
echo
echo Great work! 🥂
echo
`
pre-commit.txt:
`#!/bin/sh
. "$(dirname "$0")/_/husky.sh"

dotnet husky run -v --group "pre-commit"
`

task-runner.json
`{
  "variables": [
    {
      "name": "root-dir",
      "command": "cmd",
      "args": [ "/c", "dir", "/b" ]
    }
  ],
  "tasks": [
    {
      "name": "commit-message-linter",
      "command": "dotnet",
      "args": [ "husky", "exec", ".husky/csx/commit-lint.csx", "--args", "${args}" ]
    },
    {
      "name": "echo staged files",
      "pathMode": "absolute",
      "command": "cmd",
      "group": "pre-commit",
      "args": [ "/c", "echo", "${staged}" ]
    },
    {
      "name": "tests",
      "command": "dotnet",
      "group": "pre-commit",
      "args": [
        "test",
        "-t",
        "-v:diag",
        "--consoleLoggerParameters:ErrorsOnly",
        "--blame"
      ]
    },
    {
      "name": "dotnet-format",
      "command": "dotnet",
      "group": "pre-commit",
      "args": [ "format", "--include", "${staged}" ],
      "include": [ "**/*.cs" ]
    },
    {
      "name": "warning-check",
      "command": "dotnet",
      "group": "pre-push",
      "args": [
        "build",
        "/warnaserror"
      ],
      "include": [
        "**/*.cs"
      ]
    }
  ]
}`

commit-lint.csx

`/// <summary>
/// a simple regex commit linter example
/// https://www.conventionalcommits.org/en/v1.0.0/
/// https://github.com/angular/angular/blob/22b96b9/CONTRIBUTING.md#type
/// </summary>

using System.Text.RegularExpressions;

private var pattern = @"^(?=.{1,90}$)(?:build|feat|ci|chore|docs|fix|perf|refactor|revert|style|test)(?:\(.+\))*(?::).{4,}(?:#\d+)*(?<![\.\s])$";
private var msg = File.ReadAllLines(Args[0])[0];

if (Regex.IsMatch(msg, pattern))
   return 0;

Console.ForegroundColor = ConsoleColor.Red;
Console.WriteLine("Invalid commit message");
Console.ResetColor();
Console.WriteLine("e.g: 'feat(scope): subject' or 'fix: subject'");
Console.ForegroundColor = ConsoleColor.Gray;
Console.WriteLine("more info: https://www.conventionalcommits.org/en/v1.0.0/");

return 1;
`
